client.go

```diff
@@ -0,0 +1,105 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	bip32 "github.com/bitcoin-sv/go-sdk/compat/bip32"
+	ec "github.com/bitcoin-sv/go-sdk/primitives/ec"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/auth"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/httpx"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+)
+
+// Client provides methods for user-related and admin-related APIs.
+// This struct is designed to abstract and simplify the process of making HTTP calls
+// to the relevant endpoints. By utilizing this Client struct, developers can easily
+// interact with both user and admin APIs without needing to manage the details
+// of the HTTP requests and responses directly.
+type Client struct {
+	userAPI *user.API
+}
+
+// SharedConfig retrieves the shared configuration from the user configurations API.
+// This method constructs an HTTP GET request to the "/shared" endpoint and expects
+// a response that can be unmarshaled into the response.SharedConfig struct.
+// If the request fails or the response cannot be decoded, an error will be returned.
+func (c *Client) SharedConfig(ctx context.Context) (*response.SharedConfig, error) {
+	return c.userAPI.ConfigsAPI.SharedConfig(ctx)
+}
+
+// NewWithXPub creates a new client instance using an extended public key (xPub).
+// Generates a hierarchical deterministic (HD) key from the provided xPub and constructs
+// the necessary configuration for the API instance. The SignRequest flag is set to false,
+// indicating that requests made with this instance will not be signed.
```

Comment:
- The `HD` part is a implementation detail here and I don't see a reason to mention it in the description. 
```suggestion
// Requests made with this instance will not be signed, that's why we strongly recommend to use `WithXPriv` or `WithAccessKey` option instead. 
```

BTW. Consider to move this function to the bottom, to make sure that other developers will find `NewWithXPriv` first 🙈

Reply:
You mean the all comment or just a part of it?  From the perspective of somebody who's learning the concepts of the blockchain this comment is much more explanatory than providing simple sentence like "returns the client implementation" without any useful information. 

What's more, from the function caller perspective also without need to get into details.

Reply:
I feel like you don't get my comment - let's discuss it on PM 🙈